src/kixi/hecuba/onyx/jobs/weather_fetcher.clj

```diff
@@ -35,31 +60,57 @@
 
 (defn pull-data
   "Get Metoffice data as a string"
-  ([querydate querytime siteid]
-   (->> (client/post "http://datagovuk.cloudapp.net/query"
-                     {:form-params {:Type "Observation"
-                                    :PredictionSiteID siteid
-                                    :ObservationSiteID siteid
-                                    :Date querydate ;; dd/mm/yyyy
-                                    :PredictionTime querytime ;; 0000
-                                    }
-                      :follow-redirects true})
-        :body
-        (re-find #"https://datagovuk.blob.core.windows.net/csv/[a-z0-9]+.csv")
-        slurp
-        )))
+  [querydate querytime siteid]
+  (->> (client/post "http://datagovuk.cloudapp.net/query"
+                    {:form-params {:Type "Observation"
+                                   :PredictionSiteID siteid
+                                   :ObservationSiteID siteid
+                                   :Date querydate ;; dd/mm/yyyy
+                                   :PredictionTime querytime ;; 0000
+                                   }
+                     :follow-redirects true})
+       :body
+       (re-find #"https://datagovuk.blob.core.windows.net/csv/[a-z0-9]+.csv")
+       (client/get)
+       :body
+       ))
```

Comment:
I'd split the post and get out into separate functions. Having them bound together here will make it hard to debug, and generally a bit of a pain.